scipy/interpolate/rbf.py

```diff
@@ -188,22 +185,20 @@ def _init_function(self, r):
         return a0
 
     def __init__(self, *args, **kwargs):
-        self.xi = np.asarray([np.asarray(a, dtype=np.float_).flatten()
-                           for a in args[:-1]])
-        self.N = self.xi.shape[-1]
+        self.xi = np.asarray([np.asarray(a, dtype=np.float_).flatten() for a in args[:-1]]).T
+        self.N = self.xi.shape[0]
```

Comment:
`xi` and `N` are public attributes, they shouldn't be changed here. Looks to me like you can just add `.T` inside the calls to the distance functions instead.

Reply:
Done.